internal/controller/storageclaim_controller.go

```diff
@@ -406,13 +424,35 @@ func (r *StorageClaimReconciler) reconcilePhases() (reconcile.Result, error) {
 				if err := r.createOrReplaceVolumeSnapshotClass(volumeSnapshotClass); err != nil {
 					return reconcile.Result{}, fmt.Errorf("failed to create or update VolumeSnapshotClass: %s", err)
 				}
+			case "ClientProfile":
+				if utils.DelegateCSI {
+					clientProfile := &csiopv1a1.ClientProfile{}
+					clientProfile.Name = r.storageClaimHash
+					clientProfile.Namespace = r.OperatorNamespace
+					if _, err := controllerutil.CreateOrUpdate(r.ctx, r.Client, clientProfile, func() error {
+						if err := r.own(clientProfile); err != nil {
+							return fmt.Errorf("failed to own clientProfile resource: %v", err)
+						}
+						if err := json.Unmarshal(resource.Data, &clientProfile.Spec); err != nil {
+							return fmt.Errorf("failed to unmarshall clientProfile spec: %v", err)
+						}
```

Comment:
This PR already established a pattern where the unmarshaling happens outside and before `createOrUpdate`, can you please align here as well?

Reply:
We are unmarshaling two shapes, one is of map[string]string and another is the spec specific to the resource that we are interested and I couldn't club both.